app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,54 @@
+class RecipesController < ApplicationController
+  def index
+    @recipes = current_user.recipes
+    # @recipe = Recipe.new
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @recipe_foods = @recipe.recipe_foods
+  end
+
+  def create
+    @recipe = Recipe.new(params.require(:recipe).permit(:name, :preparation_time, :cooking_time, :description, :public, :user_id))
+    @recipe.user = current_user
+    if @recipe.save
+      redirect_to recipes_path
+    else
+      render :new
+    end
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def update
+    @recipe = Recipe.find(params[:id])
+
+    if @recipe.public.zero?
+      @recipe.update(public: 1)
+      redirect_to @recipe, notice: 'Recipe public status was successfully updated.'
+    elsif @recipe.public == 1
+      @recipe.update(public: 0)
+      redirect_to @recipe, notice: 'Recipe public status was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
```

Comment:
- [ ] Any form of manipulation of a recpie should be done by the user who created said recipe. Right now a user that did not create a recipe is able to change it from public to private, causing a security problem kindly fix this.

Video cast
[cinnamon-2023-12-22T171056+0200.webm](https://github.com/zunairkhan811/food-recipe/assets/81932478/1ecbdda0-da07-4188-bc1d-40ae5e92b926)

Reply:
@Munsa1  Thanks for notifying me. The changes have been made.